src/Controller/CRUDController.php

```diff
@@ -514,9 +514,13 @@ public function batchAction()
             $formView = $datagrid->getForm()->createView();
             $this->setFormTheme($formView, $this->admin->getFilterTheme());
 
-            // NEXT_MAJOR: Remove this line and use commented line below it instead
-            $template = $this->admin->getTemplate('batch_confirmation');
-            // $template = $this->templateRegistry->getTemplate('batch_confirmation');
+            // NEXT_MAJOR: Remove these lines and use commented lines below them instead
+            $template = !empty($batchActions[$action]['template']) ?
+                $batchActions[$action]['template'] :
+                $this->admin->getTemplate('batch_confirmation');
+            // $template = !empty($batchActions[$action]['template']) ?
+            //     $batchActions[$action]['template'] :
+            //     $this->templateRegistry->getTemplate('batch_confirmation');
```

Comment:
What are these 3 lines about?

Reply:
The commented line was there before. I just matched it to updated version.